WhaleSpotting/react-app/src/tests/components/MapChart.test.tsx

```diff
@@ -2,15 +2,45 @@ import React from "react";
 import { render, screen } from "@testing-library/react";
 import { MapChart } from "../../components/MapChart";
 import { BrowserRouter as Router } from "react-router-dom";
+import { SightingApiModel } from "../../api/models/SightingApiModel";
+import { fetchAllSightings } from "../../api/apiClient";
+
+const mockexample: SightingApiModel = {
+    id: 1,
+    sightedAt: new Date().toDateString(),
+    species: "orca",
+    quantity: 3,
+    location: "Sea",
+    longitude: 1.232,
+    latitude: 2.312,
+    description: "Whales at sea",
+    orcaType: "Orca",
+    orcaPod: "",
+    confirmed: true,
+    userId: 2,
+    username: "FakeUserConfirmed"
+};
 
 describe("Map Chart component tests", () => {
     test("Renders MapChart", () => {
-        const chosen = {id: 1, lon: 0, lat: 0};
+
+        jest.mock("../../api/apiClient", () => ({
+            __esModule: true,
+            fetchAllSightings: jest.fn(async () : Promise<SightingApiModel[]> => {
+                return Promise.resolve([mockexample]);
+            })
+        }));
+
         //eslint-disable-next-line
-        const setChosen = () => {};
-        render(<Router><MapChart chosen={chosen} setChosen={setChosen}/></Router>);
+        render(<Router><MapChart chosen={undefined} setChosen={() => {}}/></Router>);
```

Comment:
This may or may not work to not need the eslint disable
```suggestion
        render(<Router><MapChart chosen={undefined} setChosen={() => void}/></Router>);
```